src/main.h

```diff
@@ -13,6 +13,7 @@
 #ifdef ENABLE_ADDRESS_INDEXING
 #include "spentindex.h"
 #include "timestampindex.h"
+enum class AddressType;
```

Comment:
Personally I would prefer to see this forward declaration in its proper section (e.g. immediately before the declaration of `class CTransaction;` or after `class CTxInUndo;`), even though we need to add an `#ifdef` just for it, but feel free to ignore this comment.

Reply:
Done